### PR TITLE
External API - ScenesService, Scene, SceneItem, SceneFolder, Selection

### PR DESCRIPTION
**Pull request by holiber**
This must be merged after https://github.com/stream-labs/streamlabs-obs/pull/1326

### CODE REVIEW

File: app/services/api/external-api.ts [hunk 1]
```diff
@@ -14,6 +14,27 @@ export function Singleton() {
 }
 
 /**
+<<<<<<< HEAD
```
Comment (andycreeth) [inline app/services/api/external-api.ts]: There's merge resolution markup in a couple places in this file.  Kind of hilariously it's only in the comments so it's not affecting compilation.

Comment (holiber) [reply app/services/api/external-api.ts]: Ouch, looks like my IDE didn't detect conflicts in the comments, lol

File: app/services/api/external-api/scenes/scene-item.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
+import { ServiceHelper } from 'services/stateful-service';
+import { SceneItem as InternalSceneItem } from 'services/scenes';
+import { InjectFromExternalApi } from '../../external-api';
+import { ISourceModel, Source, SourcesService } from '../sources/sources';
```
Comment (andycreeth) [inline app/services/api/external-api/scenes/scene-item.ts]: absolute import

File: app/services/api/external-api/scenes/scene-item.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+import { ISceneNode, SceneNode } from './scene-node';
+import { AnchorPoint, AnchorPositions } from '../../../../util/ScalableRectangle';
```
Comment (andycreeth) [inline app/services/api/external-api/scenes/scene-item.ts]: absolute import

File: app/services/api/external-api/scenes/scene-node.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
+import { Inject } from '../../../../util/injector';
```
Comment (andycreeth) [inline app/services/api/external-api/scenes/scene-node.ts]: absolute

File: app/services/api/external-api/scenes/scene.ts [hunk 5]
```diff
@@ -0,0 +1,167 @@
+import { ServiceHelper } from 'services/stateful-service';
+import { InjectFromExternalApi } from '../../external-api';
+import { Source, SourcesService } from '../sources/sources';
```
Comment (andycreeth) [inline app/services/api/external-api/scenes/scene.ts]: Imports for this file

File: app/services/api/external-api/scenes/scenes.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import { Observable } from 'rxjs';
+import { Singleton } from '../../external-api';
+import { ScenesService as InternalScenesService } from 'services/scenes/index';
+import { ISourceAddOptions } from '../sources/sources';
+import { Inject } from '../../../../util/injector';
```
Comment (andycreeth) [inline app/services/api/external-api/scenes/scenes.ts]: Imports for this file

File: app/services/api/external-api/scenes/selection.ts [hunk 7]
```diff
@@ -0,0 +1,323 @@
+import { InjectFromExternalApi, Singleton } from '../../external-api';
+import { ServiceHelper } from 'services/stateful-service';
+import { ISceneItemActions, ISceneItemSettings, IPartialTransform } from 'services/scenes';
+import {
+  Selection as InternalSelection,
+  SelectionService as InternalSelectionService,
+} from 'services/selection';
+import { ScenesService } from './scenes';
+import { Source, SourcesService } from '../sources/sources';
+import { Inject } from '../../../../util/injector';
```
Comment (andycreeth) [inline app/services/api/external-api/scenes/selection.ts]: imports for this file

**Pull request merged**